File: tests/expectations/both/alias.c
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>
#include <uchar.h>

enum Status {
  Ok,
  Err,
};
typedef uint32_t Status;

typedef struct Dep {
  int32_t a;
  float b;
} Dep;

typedef struct Foo_i32 {
  int32_t a;
  int32_t b;
  Dep c;
} Foo_i32;

typedef Foo_i32 IntFoo;

typedef struct Foo_f64 {
  double a;
  double b;
  Dep c;
} Foo_f64;

typedef Foo_f64 DoubleFoo;

typedef int32_t Unit;

typedef Status SpecialStatus;

void root(IntFoo x, DoubleFoo y, Unit z, SpecialStatus w);

File: tests/expectations/alias.zig
const std = @import("std");

pub const Status = enum(c_int) {
    Ok = 0,
    Err = -1,
};

pub const Dep = extern struct {
    a: i32,
    b: f32,
};

pub const Foo_i32 = extern struct {
    a: i32,
    b: i32,
    c: Dep,
};

pub const IntFoo = Foo_i32;

pub const Foo_f64 = extern struct {
    a: f64,
    b: f64,
    c: Dep,
};

pub const DoubleFoo = Foo_f64;
pub const Unit = i32;
pub const SpecialStatus = Status;

extern fn root(x: IntFoo, y: DoubleFoo, z: Unit, w: SpecialStatus) anyopaque;
